server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java

```diff
@@ -583,6 +521,8 @@ public static Map<String, String> stripRealmAttributesIncludedAsFields(Map<Strin
         return a;
     }
 
+    // TODO GROUPS now that there are different mechanisms for accessing groups this could be done in a way that doesn't try to load everything
```

Comment:
Maybe we can consider as a follow-up instead of directly address in this PR? Since this PR is already quite big.

I would say we can maybe improve this for groups in the follow-up PR and add it to the groups epic https://github.com/keycloak/keycloak/issues/22261 . Also we can create separate follow-up PRs (if needed) for other objects.